pkg/auth/user/UserService.go

```diff
@@ -1294,6 +1299,129 @@ func (impl *UserServiceImpl) DeleteUser(bean *bean.UserInfo) (bool, error) {
 	return true, nil
 }
 
+// BulkDeleteUsers takes in BulkDeleteRequest and return success and error
+func (impl *UserServiceImpl) BulkDeleteUsers(request *bean.BulkDeleteRequest) (bool, error) {
+	// it handles FetchListingRequest if filters are applied will delete those users or will consider the given user ids.
+	if request.ListingRequest != nil {
+		filteredUserIds, err := impl.getUserIdsHonoringFilters(request.ListingRequest)
+		if err != nil {
+			impl.logger.Errorw("error in BulkDeleteUsers", "request", request, "err", err)
+			return false, err
+		}
+		// setting the filtered user ids here for further processing
+		request.Ids = filteredUserIds
+	}
+	err := impl.deleteUsersByIds(request)
+	if err != nil {
+		impl.logger.Errorw("error in BulkDeleteUsers", "err", err)
+		return false, err
+	}
+	return true, nil
+}
+
+// getUserIdsHonoringFilters get the filtered user ids according to the request filters and returns userIds and error(not nil) if any exception is caught.
+func (impl *UserServiceImpl) getUserIdsHonoringFilters(request *bean.FetchListingRequest) ([]int32, error) {
+	//query to get particular models respecting filters
+	query := impl.userListingRepositoryQueryBuilder.GetQueryForUserListingWithFilters(request)
+	models, err := impl.userRepository.GetAllExecutingQuery(query)
+	if err != nil {
+		impl.logger.Errorw("error while fetching user from db in GetAllWithFilters", "error", err)
+		return nil, err
+	}
+	// collecting the required user ids from filtered models
+	filteredUserIds := make([]int32, len(models))
+	for i, model := range models {
+		if !(model.EmailId == bean2.AdminUser || model.EmailId == bean2.SystemUser) {
```

Comment:
we have a method to check for devtron managed user id, use it

Reply:
DONE